lib/timespan.py

```diff
@@ -0,0 +1,89 @@
+import re
+import calendar
+from tasks.meta import OBSTimespan, current_session
+
+QUARTERS = {1: 'first',
+            2: 'second',
+            3: 'third',
+            4: 'fourth'}
+
+
+def get_timespan(timespan_id):
+    ts_id, ts_name, ts_description, ts_timespan = parse_timespan(timespan_id)
+    timespan = OBSTimespan(id=ts_id, name=ts_name, description=ts_description, timespan=ts_timespan)
+    return current_session().merge(timespan)
+
+
+def parse_timespan(timespan_id):
+    timespan_str = str(timespan_id).replace(' ', '')
+
+    ts_id, ts_name, ts_description, ts_timespan = None, None, None, None
+
+    # Year, for example '2017'
+    if re.match('\d{4}$', timespan_str):
```

Comment:
Due to we have a "closed" regexp comparison, I'd use `^\d{4}$` regexp clause to avoid parse things like `pepito2004` (all the regexp we have should start with ^)

Reply:
What about this?